### DISCUSSION TITLE
Multi material frame of a Robot

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#4090] [Q&A]
Hello,

I need some advice about complex body construction : I'll try to make the problem minimal here : 

To my understanding, to make thing correctly, we need to :  
- Create a Robot node, then draw the body of the bot using shape-like objects, placed as children of the Robot node. Let's say a simple box.
- Specify the physical properties of this body (mass....), using a Physic Node placed in the physic properties of the Robot node. Also I'm using the contactMaterial property. I set it up to "wood" for example.
- Add a plane, using it as slope (made, let's say, with concrete).
- I setup the contact between "wood" and "concrete" in WorldInfo with nice coulomb values.
That's fine, and to physic work great. I could get the box sliding of a slope at a nice speed.

- Now I want to add to the wooden body a piece of steal (physically "glued" to the wooden box, thus forming still one part), lets say to protect some corner of the body in contact with the ground. Wood and steal part will be in contact with the slope, and slide on it, so I need to add some contact properties in this. Adding an contactProperties in WorldInfo between "steal" and "concrete" is easy, but now I don't know where I should use it ! In fact the only place I could write "steal" is in the contactMaterial of the Robot, but there is already "wood" in here...

So for the moment, I'm rather stuck...

I'm very open to any feedback, in particular if I've missed some aspect of Robot building !

Thank

**Comment by omichel** [comment]
I guess you will need to add a new Solid node as a children of your robot for the steel part. This new Solid should have its `contactMaterial` property set to "steel".

**Reply by ShuffleWire** [reply]
That might be the way, but I was warned that adding Solid as children could be hazardous...
